engine/common/unit/unit.go

```diff
@@ -9,9 +9,50 @@ import (
 	"github.com/onflow/flow-go/module/irrecoverable"
 )
 
-// Unit handles synchronization management
+// Unit handles synchronization management for engines.
+// There are 2 common usage patterns:
+//
+// 1. Unit as the Component implementation for an engine:
+// Unit is used to manage the lifecycle of long-running tasks within an engine. In this case,
+// embed a component.Component interface into the parent struct, then set unit as the implementation
+// e.g.
+//
+//	type Engine struct {
+//		component.Component
+//		unit unit.Unit
```

Comment:
```suggestion
//		unit.Unit
```

I had a simpler solution like this. It works. I can launch a long running worker that consume a certain job channels like this on startup:
https://github.com/onflow/flow-go/blob/leo/ingestion-new-engine-component/engine/execution/ingestion/core.go#L81-L83

Reply:
but that also exposes the other methods like launch via the engine. I think we should avoid allowing external code to launch new work on the engine